game/songitems.cc

```diff
@@ -36,73 +42,67 @@ void SongItems::save(xmlpp::Element* songs) {
 
 SongId SongItems::addSongItem(std::string const& artist, std::string const& title, std::optional<SongId> _id) {
 	SongItem si;
+
+	if (_id.has_value() && m_songs_map.find(_id.value()) != m_songs_map.end()) {
+		si = m_songs_map.at(_id.value());
+	}
+
 	si.id = _id.value_or(assign_id_internal());
```

Comment:
assign_id_internal() would always be evaluated here, as value_or can't short circuit